app/services/wakes/pageview_count_updater_service.rb

```diff
@@ -8,14 +8,14 @@ def initialize(location)
 
   def update_pageview_count
     if pageviews_since_last_update.positive?
-      update_location_pageview_count && update_resource_aggregate_count && update_wakeable_aggregate_count
+      (update_location_pageview_count && update_resource_aggregate_count && update_wakeable_aggregate_count).tap do
+        # Yes, this will often result in two update operations
+        # Right now, code expressiveness and comprehension seem more important than performance
+        location.update(:pageview_count_checked_at => Time.zone.now)
```

Comment:
also, why do you think this timestamp (as opposed to `pageview_count_updated_through`) is where the problem is? Have you figured/reasoned that out?